### ISSUE TITLE
support for attrs?

### ISSUE THREAD

**Issue by glyph** [twisted/pydoctor#135]
It would be nice to have direct support for `attr.ib()` declarations somehow, so I don't have to take the redundant step of doing `@ivar` in the docstring.  Not quite sure how!  Perhaps attrs needs a feature (like a `doc=` argument) to make this work.

**Comment by mwhudson** [comment]
I guess you could stick something in metadata, but having doc support in attrs directly would be better for sure.

**Comment by mthuurne** [comment]
[Via Epydoc](http://epydoc.sourceforge.net/manual-docstring.html), we already support [PEP 224](https://www.python.org/dev/peps/pep-0224/) syntax:
```py
class A:
    x = 22
    """Docstring for class variable A.x"""

    def __init__(self, a):
        self.y = a
        """Docstring for instance variable A.y
```

This could be applied to attrs-based classes as well. I think the only thing that would have to change is that if an attribute is of type `attr.ib`, it should be treated as an instance variable instead of a class variable.
```py
@attr.s
class Coordinates(object):

    x = attr.ib()
    """x coordinate."""

    y = attr.ib()
    """y coordinate."""
```

Note that PEP 224 was rejected. However, since both Epydoc and pdoc support PEP 224 syntax, I think it is a de facto standard, even if it wasn't blessed by the BDFL.

**Comment by mthuurne** [comment]
A correction on my previous post: we don't actually support PEP 224 yet, since only the docstring parsing was imported from Epydoc, while PEP 224 introduces additional locations where docstrings can be found. However, if we have the ambition to provide an upgrade path for people who are currently using Epydoc, PEP 224 support would help.

**Comment by mthuurne** [comment]
I'm working on support for inline docstrings as an alternative to `@cvar` and `@ivar`. I got the class variables to a point where it passed my unit test, but then realized that the approach I took (add an `Attribute` when visiting the docstring node) wasn't great for adding type annotation support later, so I'm working on a different approach now (add an `Attribute` when visiting the assignment node and later set the docstring on the `Attribute` defined by the previous node).

**Comment by mthuurne** [comment]
Is more functionality desired or can this be closed now that #170 has been merged?

**Comment by glyph** [comment]
I think 170+175 just about does it!
